File: ingraph-bulk-loader/src/test/java/ingraph/bulkloader/csv/TrainBenchmarkTest.java
package ingraph.bulkloader.csv;

import java.io.IOException;
import java.util.Arrays;
import java.util.Collection;
import java.util.List;
import java.util.Map;

import org.junit.Test;
import org.neo4j.driver.v1.types.Node;
import org.neo4j.driver.v1.types.Relationship;
import org.supercsv.prefs.CsvPreference;

import com.google.common.collect.ImmutableMap;

import ingraph.bulkloader.csv.loader.MassCsvLoader;
import neo4j.driver.util.PrettyPrinter;

public class TrainBenchmarkTest {

	private static final String PREFIX = "../graphs/trainbenchmark/railway-repair-1-";
	private static final String POSTFIX = ".csv";

	final Map<String, Collection<String>> nodeFilenames = ImmutableMap.<String, Collection<String>>builder() //
			.put(PREFIX + "Region" + POSTFIX, Arrays.asList("Region")) //
			.put(PREFIX + "Route" + POSTFIX, Arrays.asList("Route")) //
			.put(PREFIX + "Segment" + POSTFIX, Arrays.asList("Segment", "TrackElement")) //
			.put(PREFIX + "Semaphore" + POSTFIX, Arrays.asList("Semaphore")) //
			.put(PREFIX + "Sensor" + POSTFIX, Arrays.asList("Sensor")) //
			.put(PREFIX + "Switch" + POSTFIX, Arrays.asList("Switch", "TrackElement")) //
			.put(PREFIX + "SwitchPosition" + POSTFIX, Arrays.asList("SwitchPosition")) //
			.build();

	final Map<String, String> relationshipFilenames = ImmutableMap.<String, String>builder() //
			.put(PREFIX + "connectsTo" + POSTFIX, "connectsTo") //
			.put(PREFIX + "entry" + POSTFIX, "entry") //
			.put(PREFIX + "exit" + POSTFIX, "exit") //
			.put(PREFIX + "follows" + POSTFIX, "follows") //
			.put(PREFIX + "monitoredBy" + POSTFIX, "monitoredBy") //
			.put(PREFIX + "requires" + POSTFIX, "requires") //
			.put(PREFIX + "target" + POSTFIX, "target") //
			.build();

	@Test
	public void testLoad() throws IOException {
		final MassCsvLoader mcl =  new MassCsvLoader(nodeFilenames, relationshipFilenames, CsvPreference.STANDARD_PREFERENCE);
		List<Node> nodes = mcl.getNodes();
		List<Relationship> relationships = mcl.getRelationships();

		for (int i = 0; i < Math.min(10, nodes.size()); i++) {
			System.out.println(PrettyPrinter.toString(nodes.get(i)));
		}

		for (int i = 0; i < Math.min(10, relationships.size()); i++) {
			System.out.println(PrettyPrinter.toString(relationships.get(i)));
		}
	}

}

File: tests/src/main/scala/ingraph/tests/TrainBenchmarkTest.scala
package ingraph.tests

import org.scalatest.{FunSuite, Ignore}

import scala.io.Source

@Ignore
abstract class TrainBenchmarkTest extends FunSuite {

  def modelPath(entityName: String) = s"../graphs/trainbenchmark/railway-repair-2-${entityName}.csv"

  def queryPath(query: String): String = s"../queries/trainbenchmark/${query}.cypher"

  //def queryResultPath(workload: String, query: Int): String = queryPath(workload, query).dropRight(".cypher".length) + "-50.bin"

  val nodeFilenames: Map[String, List[String]] = Map(
    modelPath("Region") -> List("Region"),
    modelPath("Route") -> List("Route"),
    modelPath("Segment") -> List("Segment", "TrackElement"),
    modelPath("Semaphore") -> List("Semaphore"),
    modelPath("Sensor") -> List("Sensor"),
    modelPath("Switch") -> List("Switch", "TrackElement"),
    modelPath("SwitchPosition") -> List("SwitchPosition")
  )

  val relationshipFilenames: Map[String, String] = Map(
    modelPath("connectsTo") -> "connectsTo",
    modelPath("entry") -> "entry",
    modelPath("exit") -> "exit",
    modelPath("follows") -> "follows",
    modelPath("monitoredBy") -> "monitoredBy",
    modelPath("requires") -> "requires",
    modelPath("target") -> "target"
  )

  case class TestCase(name: String)

  val testCases = List("PosLength").map(new TestCase(_))

  testCases.filter(_ != null) //
    .foreach(
    t =>
      test(t.name) {

      val querySpecification = Source.fromFile(queryPath(t.name)).getLines().mkString("\n")

      runQuery(t.name, querySpecification)
    })

  def runQuery(queryName: String, querySpecification: String)
}
